storm-core/src/clj/backtype/storm/cluster.clj

```diff
@@ -489,10 +489,10 @@
         (let [path (backpressure-path storm-id node port)
               existed (exists-node? cluster-state path false)]
           (if existed
-            (if (not on?)
-              (delete-node cluster-state path))   ;; delete the znode since the worker is not congested
-            (if on?
-              (set-ephemeral-node cluster-state path nil acls))))) ;; create the znode since worker is congested
+            (when (not on?)
```

Comment:
Since there is one conditional statement to run, is there any special reason for replacing "if" with "when"?